File: web-app/src/app/norm/entity/Teacher.ts
/**
 * 教师
 */
export class Teacher {
  id: number;
  name: string;
  username: string;
  email: string;
  sex: boolean;
  createTime: number;

  /**
   * 构造函数
   * @param id id
   * @param username 用户名
   * @param name 姓名
   * @param email 邮箱
   * @param sex 性别
   */
  constructor(id: number, username: string, name: string, email?: string, sex?: boolean) {
    this.id = id;
    this.name = name;
    this.username = username;
    this.email = email;
    this.sex = sex;
  }
}

File: api/src/main/java/com/mengyunzhi/springbootstudy/entity/Teacher.java
package com.mengyunzhi.springbootstudy.entity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

/**
 * 教师模型，用于更方便的存储查表后返回的数据
 */
@Entity
public class Teacher {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private Boolean sex;
    private String username;
    private String email;
    private Long createTime;
    private Long updateTime;
    private String password = "yunzhi";

    public Teacher() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Boolean getSex() {
        return sex;
    }

    public void setSex(Boolean sex) {
        this.sex = sex;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Long getCreateTime() {
        return createTime;
    }

    public void setCreateTime(Long createTime) {
        this.createTime = createTime;
    }

    public Long getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(Long updateTime) {
        this.updateTime = updateTime;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }
}
